### PR TITLE
Support appservice with java11

### PR DESCRIPTION
**Pull request by Flanker32**
Add support for java 11 app service, based on https://github.com/Azure/azure-libraries-for-java/pull/715

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class RuntimeStackUtils {
+
+    public static final String DEFAULT_JAVA_VERSION = "java11";
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: name conflict of DEFAULT_JAVA_VERSION = JavaVersion.JAVA_8_NEWEST;

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            if (getJavaVersionFromRuntimeStack(runtimeStack).equalsIgnoreCase(javaVersion) &&
+                !runtimeStack.stack().equals("JAVA")) {
+                result.add(getWebContainerFromRuntimeStack(runtimeStack));
+            }
+        }
+        result.add(javaVersion);
+        return new ArrayList<>(result);
+    }
+
+    public static List<String> getValidJavaVersion() {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: getValidJavaVersions

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        final Set<String> result = new HashSet<>();
+        for (final RuntimeStack runtimeStack : getValidRuntimeStacks()) {
+            result.add(getJavaVersionFromRuntimeStack(runtimeStack));
+        }
+        return new ArrayList<>(result);
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: use lambda for simplicity

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            }
+        }
+        return null;
+    }
+
+    public static List<RuntimeStack> getValidRuntimeStacks() {
+        return new ArrayList<>(runtimeStacks);
+    }
+
+    public static List<String> getValidWebContainer(String javaVersion) {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: use lambda&distinct for simplicity

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            if (runtimeStack.stack().equals("JAVA") &&
+                getJavaVersionFromRuntimeStack(runtimeStack).equalsIgnoreCase(javaVersion)) {
+                return runtimeStack;
+            }
+        }
+        return null;
+    }
+
+    public static RuntimeStack getRuntimeStack(String javaVersion, String webContainer) {
+        if (StringUtils.isEmpty(webContainer) || getValidJavaVersion().contains(webContainer)) {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: can we remove the second condition getValidJavaVersion().contains(webContainer) ?

RuntimeStackUtils.getRuntimeStack("jre8", "jre8") looks weird

Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: `azure-webapp:config` will generate both javaVersion and runtime, so add the condition here

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java (hunk 1 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: Change name to default linux java version and set both windows and linux default value to java11

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java (hunk 2 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: fixed

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java (hunk 3 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: webapp plugin use jdk1.7 for some users use java7 as webapp rutime

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java (hunk 4 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/RuntimeStackUtils.java]: webapp plugin use jdk1.7 for some users use java7 as webapp rutime

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppConfiguration.java [hunk 6]
```diff
@@ -27,7 +26,7 @@
 
     public static final Region DEFAULT_REGION = Region.EUROPE_WEST;
     public static final PricingTierEnum DEFAULT_PRICINGTIER = PricingTierEnum.P1V2;
-    public static final JavaVersion DEFAULT_JAVA_VERSION = JavaVersion.JAVA_ZULU_1_8_0_144;
+    public static final JavaVersion DEFAULT_JAVA_VERSION = JavaVersion.JAVA_8_NEWEST;
```
[the remark this answers is not in the document]
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppConfiguration.java]: has changed to Java 11

**Pull request merged**